### ISSUE TITLE
[Bug Report]: Flashing through WSL1 environment not working since v0.6.0 (VSC-549)

### ISSUE THREAD

**Issue by risinek** [espressif/vscode-esp-idf-extension#295]
**Pre Bug Report Checklist**
Before reporting any bug please make sure of these points.

- [x] Make sure you have searched for existing bugs and features request before you post an issue.
- [x] This is a bug report for the ESP-IDF Visual Studio Code extension and **not** an ESP-IDF bug report.
- [x] I've read the docs and found no information that could have helped solving the issue.

**Describe the bug**
Since updating to v0.6.0 the extension (that is installed in WSL debian and runs in VSC installed on Win 10) is fetching serial ports from Windows 10 instead of WSL1. .
On version v.0.5.1 it shows `/dev/ttySx` ports correctly from WSL environment. Since v0.6.0 I get `COMx` from Windows environment.

When I try to flash the project, it runs the flash process via `powershell.exe` which does not exist in WSL Linux environment. (I have also tried manually overriding `idf.port` in `settings.json` to `"idf.port": "/dev/ttyS4"` instead of `COM4` that was provided via UI serial port selector with the same result)
```
Executing task: powershell.exe -Command "python \\\\wsl$\\Debian\\home\\username\\esp\\esp-idf\\components\\esptool_py\\esptool\\esptool.py -p /dev/ttyS4 -b 460800 --after hard_reset write_flash --flash_mode dio --flash_freq 40m --flash_size detect 0x8000 partition_table\partition-table.bin 0x1000 bootloader\bootloader.bin 0x10000 app.bin"
```
As I don't have python in Windows environment installed, this will end up in message 
```
Python was not found; run without arguments to install from the Microsoft Store,
 or disable this shortcut from Settings > Manage App Execution Aliases.
```

I'm able to build project via WSL environment, so this applies only for selecting serial port and flashing.

**To Reproduce**

1. Have WSL1 installed
2. Run VSC project in Remote-WSL environment
3. Build project
4. Select serial port of ESP32 connected via USB
**Expected behavior:** it will give you Linux `/dev/ttySx` serial ports to choose from
**Actual behavior:** it will give Windows COMs instead
5. Flash project via USB
**Expected behavior:** it will run python directly from WSL environment
**Actual behavior:** it will run powershell.exe in WSL environment

**Expected behavior**
ESP-IDF extension should use Linux environment only tools if it's running inside Remote-WSL environment in VSC.

**Screenshots**
Not needed. I can provide some if demanded.

**Environment (please complete the following information):**

- OS Name & Version: Windows 10 1909
- VSCode Version: 1.52.1 (installed on WSL Debian)
- ESP-IDF Version: 4.1
- Python Version: 3.5
- WSL - WSL1 Debian
- Remote-WSL - remote-wsl-0.52.0

**Logs**
If applicable, please share the log file which can be obtained from

```
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)","level":"error","timestamp":"2021-01-16T10:31:22.025Z"}
{"message":"ESP-IDF Terminal process ended with exit code 1.","stack":"Error: ESP-IDF Terminal process ended with exit code 1.\n\tat ChildProcess.<anonymous> (/home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:258814)\n\tat ChildProcess.emit (events.js:223:5)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)","level":"error","timestamp":"2021-01-16T10:31:47.799Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Socket.<anonymous> (internal/child_process.js:430:11)\n\tat Socket.emit (events.js:223:5)\n\tat Pipe.<anonymous> (net.js:664:12)","level":"error","timestamp":"2021-01-16T10:32:24.142Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Socket.<anonymous> (internal/child_process.js:430:11)\n\tat Socket.emit (events.js:223:5)\n\tat Pipe.<anonymous> (net.js:664:12)","level":"error","timestamp":"2021-01-16T10:37:59.130Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)","level":"error","timestamp":"2021-01-16T10:40:41.486Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)","level":"error","timestamp":"2021-01-16T10:40:42.552Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Socket.<anonymous> (internal/child_process.js:430:11)\n\tat Socket.emit (events.js:223:5)\n\tat Pipe.<anonymous> (net.js:664:12)","level":"error","timestamp":"2021-01-16T10:49:34.094Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)","level":"error","timestamp":"2021-01-16T10:49:35.695Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)","level":"error","timestamp":"2021-01-16T10:51:25.678Z"}
{"user":true,"message":"Failed to flash because of some unusual error","stack":"Error: Task ESP-IDF Flash exited with code 1\n\tat Function.<anonymous> (/home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:47342)\n\tat Generator.next (<anonymous>)\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:45946\n\tat new Promise (<anonymous>)\n\tat o (/home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:45691)\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:47206\n\tat l.fire (/home/username/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/out/vs/server/remoteExtensionHostProcess.js:46:801)\n\tat _.$onDidEndTaskProcess (/home/username/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/out/vs/server/remoteExtensionHostProcess.js:954:168)\n\tat processTicksAndRejections (internal/process/task_queues.js:94:5)","level":"error","timestamp":"2021-01-16T10:51:28.926Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)","level":"error","timestamp":"2021-01-16T10:59:00.422Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Socket.<anonymous> (internal/child_process.js:430:11)\n\tat Socket.emit (events.js:223:5)\n\tat Pipe.<anonymous> (net.js:664:12)","level":"error","timestamp":"2021-01-16T11:00:18.723Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Socket.<anonymous> (internal/child_process.js:430:11)\n\tat Socket.emit (events.js:223:5)\n\tat Pipe.<anonymous> (net.js:664:12)","level":"error","timestamp":"2021-01-16T11:11:03.388Z"}
{"message":"Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.10.0-esp32-20191114 (2019-11-14-14:15)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:7658\n\tat ChildProcess.exithandler (child_process.js:286:7)\n\tat ChildProcess.emit (events.js:223:5)\n\tat maybeClose (internal/child_process.js:1021:16)\n\tat Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)","level":"error","timestamp":"2021-01-16T11:11:11.262Z"}
{"user":true,"message":"Failed to flash because of some unusual error","stack":"Error: Task ESP-IDF Flash exited with code 1\n\tat Function.<anonymous> (/home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:47342)\n\tat Generator.next (<anonymous>)\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:45946\n\tat new Promise (<anonymous>)\n\tat o (/home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:45691)\n\tat /home/username/.vscode-server/extensions/espressif.esp-idf-extension-0.6.0/dist/extension.js:1:47206\n\tat l.fire (/home/username/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/out/vs/server/remoteExtensionHostProcess.js:46:801)\n\tat _.$onDidEndTaskProcess (/home/username/.vscode-server/bin/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/out/vs/server/remoteExtensionHostProcess.js:954:168)\n\tat processTicksAndRejections (internal/process/task_queues.js:94:5)","level":"error","timestamp":"2021-01-16T11:11:13.776Z"}
```

**Extension Configuration settings**
```
{
    "idf.espIdfPath": "/home/username/esp/esp-idf",
    "idf.toolsPath": "/home/username/.espressif",
    "idf.pythonBinPath": "/home/username/.espressif/python_env/idf4.1_py3.5_env/bin/python",
    "idf.customExtraPaths": "/home/username/.espressif/python_env/idf4.1_py3.5_env/bin:/usr/bin:/home/username/.espressif/tools/xtensa-esp32-elf/esp-2020r2-8.2.0/xtensa-esp32-elf/bin:/home/username/.espressif/tools/xtensa-esp32s2-elf/esp-2020r2-8.2.0/xtensa-esp32s2-elf/bin:/home/username/.espressif/tools/esp32ulp-elf/2.28.51-esp-20191205/esp32ulp-elf-binutils/bin:/home/username/.espressif/tools/esp32s2ulp-elf/2.28.51-esp-20191205/esp32s2ulp-elf-binutils/bin:/home/username/.espressif/tools/openocd-esp32/v0.10.0-esp32-20191114/openocd-esp32/bin",
    "idf.customExtraVars": "{\"OPENOCD_SCRIPTS\":\"/home/username/.espressif/tools/openocd-esp32/v0.10.0-esp32-20191114/openocd-esp32/share/openocd/scripts\"}",
    "idf.port": "COM4",
    "idf.flashType": "UART"
}
```

**Additional context**
Downgrading to v0.5.1 fixed the problem and I can flash again.

My guess is this is related to a PR https://github.com/espressif/vscode-esp-idf-extension/pull/224

**Comment by brianignacio5** [comment]
Thanks for reporting this. We tried to fix the issue in WSL 2 of serial ports are not available by using windows host python instead of Linux tools as explained in [WSL doc](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/WSL.md). I can see how this affect WSL 1 behavior. Will fix soon to avoid using this method on WSL 1.

**Comment by risinek** [comment]
I just tried to update to v0.6.1, but unfortunately I still run into the same issue - if I try to flash via WSL1, it tries to run powershell.exe and fails due to python missing in Windows (wsl host). 
If there is anything I can help you with in term of testing/providing more info about my setup, let me know.

**Comment by risinek** [comment]
I've just checked your PR again, and I see the condition whether WSL1 or WSL2 should be expected depends on OS release right? But in fact, you can have WSL1 and WSL2 side by side on newest Windows and switch between them as you wish. 
My idea is to introduce some setting/config property and set WSL2 approach as default, and if user wants to use WSL1 on purpose, he would just override it in the settings.json file or somewhere. Do you think this is feasible @brianignacio5?

**Comment by brianignacio5** [comment]
Are you sure you are calling the right distribution in Visual Studio Code ? Make sure to use `Remote-WSL: New Window using distro...` instead of `Remote-WSL: New Window...` since the latter will use default WSL distribution. I've tested this with both WSL 1 and WSL 2 and they work properly, as in WSL 1 uses ports /dev/ttyS and WSL 2 calls Powershell.exe

**Comment by risinek** [comment]
I have never updated to WSL2 nor enabled anything related to it, but still 0.6.1 and now 1.0.0 also still try to flash using powershell.exe and it's looking for python on Windows. Do you have some hints what info I should gather to debug why the condition for WSL1/WSL2 still resolves my WSL env as WSL2?

**Comment by brianignacio5** [comment]
In the WSL terminal, type `uname -r` and let me know the result.

Also in Windows do `wsl -l -v` from Powershell terminal.

We use the fact that kernel > 4.19 should be WSL 2 as stated in https://askubuntu.com/a/1177730

**Comment by risinek** [comment]
Alright here are the outputs from those commands.
```
$ uname -r
4.4.0-18362-Microsoft
```

```
> wsl -l -v
  NAME      STATE           VERSION
* Debian    Running         1
 ```
So clearly I have WSL1 but the extension still tries to go with the workaround for WSL2 while flashing.

**Comment by risinek** [comment]
I think I found the problem - in your PR #298 you didn't updated the condition in `flash/flashTask.ts` [(block between lines 82 and 91) ](https://github.com/brianignacio5/vscode-esp-idf-extension/blob/6c1d1017e261665ccdd7c1cd6be7d7b6d90ca13d/src/flash/flashTask.ts#L82..L91)
Should I fix it and open PR or will you do it?

**Comment by brianignacio5** [comment]
> I think I found the problem - in your PR #298 with the fix you didn't include the condition in `flash/flashTask.ts` [(block between lines 82 and 91) ](https://github.com/brianignacio5/vscode-esp-idf-extension/blob/6c1d1017e261665ccdd7c1cd6be7d7b6d90ca13d/src/flash/flashTask.ts#L82..L91)
> I've opened PR with a fix #331 331

Oh wow ! Thanks for looking into it @risinek and your PR 😄
